### ISSUE TITLE
Add support for color schemes

### ISSUE THREAD

**Issue by xmstspider** [iptraf-ng/iptraf-ng#1]
Please add support for color schemes. I am using iptraf-ng on terminal using dark pastel colors and the default color scheme is almost unreadable. The only workaround is to disable colors.

Would it be possible to add support for configuring colors using either configuration menu och external files like vim or mc?

Thanks

**Comment by pecameron** [comment]
On 05/08/2018 02:57 AM, Zdenek Sedlak wrote:
>
> Please add support for color schemes. I am using iptraf-ng on terminal 
> using dark pastel colors and the default color scheme is almost 
> unreadable. The only workaround is to disable colors.
>
> Would it be possible to add support for configuring colors using 
> either configuration menu och external files like vim or mc?
>
Could you create a bug for this request?
thnakd,
phil

**Comment by xmstspider** [comment]
Sure, thanks. Do you mean in fedoraproject?

**Comment by pecameron** [comment]
@xmstspider yes fedora. Sorry about the delay in this reply. I have been busy with other projects.

**Comment by vsamel** [comment]
test

**Comment by mrmeszaros** [comment]
Could You please link the fedora issue / issue-tracker?

I am also interested in a dark mode (or other color schemes) for the tool.

**Comment by mb720** [comment]
>The only workaround is to disable colors.

How do you disable colors in iptraf?

**Comment by kingsly** [comment]
Workaround for folks stuck on terminals where iptraf-ng default colours make it unusable.

You can disable colours/make it monochrome 

C(o)nfigure - > (C)olor  ; exit and restart iptraf-ng. 

Edit: FWIW the config file is stored at /var/lib/iptraf-ng/iptraf.cfg (at least on Debian/derivatives)

**Comment by unixbhaskar** [comment]
> Edit: FWIW the config file is stored at /var/lib/iptraf-ng/iptraf.cfg

I am on Gentoo, and it is a binary file :(  

Now, how can I change the color scheme to solarized-dark??

**Comment by kingsly** [comment]
Sadly iptraf-ng only switches between colour and monochome. 

The colour scheme is hardcoded /mapped to static color names defined in `curses.h`

So only way to "fix" iptraf-ng is to change the fg/bg pairs defined in `src/deskman.c`

I use this wrapper that overrides the terminal's colors and makes iptraf-ng usable  with solarized-dark and resets to default upon exist.

Known issue: 

 iptraf-ng randomly goes back to the default colors midway during some menu operations and  blinds me with the splash of unreadable colours.. It happens quite often but not easy to replicate, haven't figured out if it's  iptraf-ng sending some kind of reset signal or my terminal reacting to my keystrokes.  YMMV

```
#!/bin/sh

#Override colours to make iptraf-ng readable in solarized dark
printf '\033]4;4;#002b36\a' #green/"blue" terminal colour 4 
printf '\033]4;7;#002b36\a' #green/"white" terminal colour 7 
printf '\033]4;0;#ffff90\a' #"white"/"black" terminal colour 0

sudo iptraf-ng

#Reset colours back to solarized dark.
#printf '\033]4;4;#268bd2\a' #"Blue"
#printf '\033]4;7;#eee8d5\a' #"White"
#printf '\033]4;0;#073642\a' #"Black"

#more efficient reset for modern terminals
printf '\033]104\a'

```


<img width="1282" height="1024" alt="Image" src="https://github.com/user-attachments/assets/78adee29-8a07-4962-a6c2-06221e993408" />

**Comment by unixbhaskar** [comment]
> Sadly iptraf-ng only switches between colour and monochome.
> 
> The colour scheme is hardcoded /mapped to static color names defined in `curses.h`
> 
> So only way to "fix" iptraf-ng is to change the fg/bg pairs defined in `src/deskman.c`
> 
> I use this wrapper that overrides the terminal's colors and makes iptraf-ng usable with solarized-dark and resets to default upon exist.
> 
> Known issue:
> 
> iptraf-ng randomly goes back to the default colors midway during some menu operations and blinds me with the splash of unreadable colours.. It happens quite often but not easy to replicate, haven't figured out if it's iptraf-ng sending some kind of reset signal or my terminal reacting to my keystrokes. YMMV
> 
> ```
> #!/bin/sh
> 
> #Override colours to make iptraf-ng readable in solarized dark
> printf '\033]4;4;#002b36\a' #green/"blue" terminal colour 4 
> printf '\033]4;7;#002b36\a' #green/"white" terminal colour 7 
> printf '\033]4;0;#ffff90\a' #"white"/"black" terminal colour 0
> 
> sudo iptraf-ng
> 
> #Reset colours back to solarized dark.
> #printf '\033]4;4;#268bd2\a' #"Blue"
> #printf '\033]4;7;#eee8d5\a' #"White"
> #printf '\033]4;0;#073642\a' #"Black"
> 
> #more efficient reset for modern terminals
> printf '\033]104\a'
> ```
> 
> <img alt="Image" width="1282" height="1024" src="https://private-user-images.githubusercontent.com/467654/573603409-78adee29-8a07-4962-a6c2-06221e993408.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzUyNDg1ODUsIm5iZiI6MTc3NTI0ODI4NSwicGF0aCI6Ii80Njc2NTQvNTczNjAzNDA5LTc4YWRlZTI5LThhMDctNDk2Mi1hNmMyLTA2MjIxZTk5MzQwOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwM1QyMDMxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMWYxZWQ5NThmOTFmYjNiNzdlYWNjOWRlMDYxOWY0Y2ExN2QxMjU1OGYxOWU1Mjc2NjQxMDM2MDFjNTliY2MwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oVL1R1VbOoqaTqD43SQc5GU6wi_0DKC5i-c4z50FD3Y">

Oh, bloody good!! Thanks a bunch for the effort and it works.  Yet to face the glitch....
